lite/helpers.go

```diff
@@ -115,47 +113,46 @@ func makeVote(header *types.Header, vals *types.ValidatorSet, key crypto.PrivKey
 	return vote
 }
 
-// Silences warning that vals can also be merkle.Hashable
-// nolint: interfacer
 func genHeader(chainID string, height int64, txs types.Txs,
-	vals *types.ValidatorSet, appHash, consHash, resHash []byte) *types.Header {
+	valset, nextValset *types.ValidatorSet, appHash, consHash, resHash []byte) *types.Header {
 
 	return &types.Header{
 		ChainID:  chainID,
 		Height:   height,
-		Time:     time.Now(),
+		Time:     time.Now().Round(0).UTC(),
 		NumTxs:   int64(len(txs)),
 		TotalTxs: int64(len(txs)),
 		// LastBlockID
 		// LastCommitHash
-		ValidatorsHash:  vals.Hash(),
-		DataHash:        txs.Hash(),
-		AppHash:         appHash,
-		ConsensusHash:   consHash,
-		LastResultsHash: resHash,
+		ValidatorsHash:     valset.Hash(),
+		NextValidatorsHash: nextValset.Hash(),
+		DataHash:           txs.Hash(),
+		AppHash:            appHash,
+		ConsensusHash:      consHash,
+		LastResultsHash:    resHash,
 	}
 }
 
-// GenCommit calls genHeader and signHeader and combines them into a Commit.
-func (v ValKeys) GenCommit(chainID string, height int64, txs types.Txs,
-	vals *types.ValidatorSet, appHash, consHash, resHash []byte, first, last int) Commit {
+// GenSignedHeader calls genHeader and signHeader and combines them into a SignedHeader.
+func (pkz privKeys) GenSignedHeader(chainID string, height int64, txs types.Txs,
```

Comment:
Maybe better to accept header and commit as parameters? It will read better.

Reply:
TODO make issue